hmftools.wdl

```diff
@@ -0,0 +1,927 @@
+                      category: "advanced"}
+    }
+}
+
+task GripssHardFilterApplicationKt {
+    input {
+        File inputVcf
+        String outputPath = "gripss_hard_filter.vcf.gz"
+
+        String memory = "3G"
+        String javaXmx = "2G"
+        Int timeMinutes = 15
+        String dockerImage = "quay.io/biocontainers/hmftools-gripss:1.11--hdfd78af_0"
+    }
+
+    command {
+        java -Xmx~{javaXmx} -XX:ParallelGCThreads=1 \
+        -cp /usr/local/share/hmftools-gripss-1.11-0/gripss.jar \
+        com.hartwig.hmftools.gripss.GripssHardFilterApplicationKt \
+        -input_vcf ~{inputVcf} \
+        -output_vcf ~{outputPath} 
+    }
+
+    output {
+        File outputVcf = outputPath
+        File outputVcfIndex = outputPath + ".tbi"
+    }
+
+    runtime {
+        memory: memory
+        time_minutes: timeMinutes # !UnknownRuntimeKey
+        docker: dockerImage
+    }
+
+    parameter_meta {
+        inputVcf: {description: "The input VCF.", category: "required"}
+        outputPath: {description: "The path where th eoutput VCF will be written.", category: "common"}
+        memory: {description: "The amount of memory this job will use.", category: "advanced"}
+        javaXmx: {description: "The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.",
+                  category: "advanced"}
+        timeMinutes: {description: "The maximum amount of time the job will run in minutes.", category: "advanced"}
+        dockerImage: {description: "The docker image used for this task. Changing this may result in errors which the developers may choose not to address.",
+                      category: "advanced"}
+    }
+}
+
+task HealthChecker {
+    input {
+        String outputDir = "."
+        String normalName
+        File normalFlagstats
+        File normalMetrics
+        String tumorName
+        File tumorFlagstats
+        File tumorMetrics
+        Array[File]+ purpleOutput
+
+        String javaXmx = "2G"
+        String memory = "1G"
+        Int timeMinutes = 1
+        String dockerImage = "quay.io/biowdl/health-checker:3.2"
+    }
+
+    command {
+        set -e
+        mkdir -p ~{outputDir}
+        health-checker -Xmx~{javaXmx} -XX:ParallelGCThreads=1 \
+        -reference ~{normalName} \
+        -ref_flagstat_file ~{normalFlagstats} \
+        -ref_wgs_metrics_file ~{normalMetrics} \
+        -tumor ~{tumorName} \
+        -tum_flagstat_file ~{tumorFlagstats} \
+        -tum_wgs_metrics_file ~{tumorMetrics} \
+        -purple_dir ~{sub(purpleOutput[0], basename(purpleOutput[0]), "")} \
+        -output_dir ~{outputDir}   
+    }
+ 
+
+    output {
+        File? healthCheckSucceeded = "~{outputDir}/~{tumorName}.HealthCheckSucceeded"
+        File? healthCheckFailed = "~{outputDir}/~{tumorName}.HealthCheckFailed"
```

Comment:
Is there something of use in these files? Or can we do some WDL magic and return a Boolean? That would be a more convenient if the files are empty.

If there is information in the files we could do
```WDL
output {
    Boolean healthCheckSucceeded = (expression)
    File outputFile = (appropriate file)
}
```
Given the output I think there is always 1 file in the output, so it might be more convenient to return it trough non-optional means.

Reply:
Hmm, I believe these files should be empty, but I'm not sure how to achieve this in WDL. I don't think there is a `exists` function.

Reply:
Ah. Then it really is a boolean. 
You could use `read_boolean()` from WDL and in the script use the `test` function `test -e /path/to/file` to echo a result that is piped into a file.